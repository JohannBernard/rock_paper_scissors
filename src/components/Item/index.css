.my_coach_item{
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  width: 30vw;
  border: 4px black solid;
  margin: 32px 5%;
  border-radius: 32px;
  box-shadow: 0 2px 4px 0 rgb(0 0 0 / 20%), 0 25px 50px 0 rgb(0 0 0 / 10%);
  padding: 16px 32px 32px;
  transition: all 0.2s;
}

.my_coach_item_win{
  background-color: rgba(75, 181, 67, 0.2);
}
.my_coach_item_lost{
  background-color: rgba(220, 53, 69, 0.2);
}

.my_coach_item_icon {
  font-size: 96px;
  margin: 0;
}

.my_coach_item_label{
  line-height: 48px;
  font-size: 48px;
  font-weight: 600;
  margin: 0;
}

/* Responsive */
@media screen and (max-width: 800px) {
  .my_coach_item{
    margin: 16px;
    border-radius: 16px;
    padding: 16px;
  }

  .my_coach_item_icon {
    font-size: 48px;
  }
  
  .my_coach_item_label{
    line-height: 28px;
    font-size: 28px;
  }
}