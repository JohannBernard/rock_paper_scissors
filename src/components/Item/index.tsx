import './index.css'
import IItem from '../../interfaces/IItem'
import GameHelper from '../../helpers/GameHelper'

interface IItemProps {
    item?: IItem
    opponent?: IItem
    id?: number
}

function Item(props: IItemProps) {
    const wonStatus = GameHelper.wonStatus(props.item, props.opponent)
    return props.item ? (
        <div
            className={
                'my_coach_item' +
                (wonStatus === 1 ? ' my_coach_item_win' : '') +
                (wonStatus === -1 ? ' my_coach_item_lost' : '')
            }
            data-testid={props.id ? 'testItem-' + props.id : ''}
        >
            <p
                className="my_coach_item_icon"
                data-testid={props.id ? 'testItemIcon-' + props.id : ''}
            >
                {props.item.icon}
            </p>
            <p
                className="my_coach_item_label"
                data-testid={props.id ? 'testItemLabel-' + props.id : ''}
            >
                {props.item.label}
            </p>
        </div>
    ) : (
        <></>
    )
}

export default Item
