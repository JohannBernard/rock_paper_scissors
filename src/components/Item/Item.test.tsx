import { render, screen, cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'
import Item from './index'

afterEach(() => {
    cleanup()
})

test('should render rock item component', () => {
    const rockData = {
        name: 'rock',
        label: 'Pierre',
        icon: 'ðŸ‘Š',
        strongerThan: ['scissors'],
    }
    render(<Item item={rockData} id={1} />)
    const itemElement = screen.getByTestId('testItem-1')
    expect(itemElement).toBeInTheDocument()
    const itemIconElement = screen.getByTestId('testItemIcon-1')
    expect(itemIconElement).toBeInTheDocument()
    expect(itemIconElement).toHaveTextContent('ðŸ‘Š')
    const itemLabelElement = screen.getByTestId('testItemLabel-1')
    expect(itemLabelElement).toHaveTextContent('Pierre')
})

test('should render paper item component', () => {
    const paperData = {
        name: 'paper',
        label: 'Feuille',
        icon: 'âœ‹',
        strongerThan: ['rock'],
    }
    render(<Item item={paperData} id={2} />)
    const itemElement = screen.getByTestId('testItem-2')
    expect(itemElement).toBeInTheDocument()
    const itemIconElement = screen.getByTestId('testItemIcon-2')
    expect(itemIconElement).toBeInTheDocument()
    expect(itemIconElement).toHaveTextContent('âœ‹')
    const itemLabelElement = screen.getByTestId('testItemLabel-2')
    expect(itemLabelElement).toHaveTextContent('Feuille')
})

test('should render scissors item component', () => {
    const scissorsData = {
        name: 'scissors',
        label: 'Ciseaux',
        icon: 'âœŒ',
        strongerThan: ['paper'],
    }
    render(<Item item={scissorsData} id={3} />)
    const itemElement = screen.getByTestId('testItem-3')
    expect(itemElement).toBeInTheDocument()
    const itemIconElement = screen.getByTestId('testItemIcon-3')
    expect(itemIconElement).toBeInTheDocument()
    expect(itemIconElement).toHaveTextContent('âœŒ')
    const itemLabelElement = screen.getByTestId('testItemLabel-3')
    expect(itemLabelElement).toHaveTextContent('Ciseaux')
})

test('matches item snapshot', () => {
    const rockData = {
        name: 'rock',
        label: 'Pierre',
        icon: 'ðŸ‘Š',
        strongerThan: ['scissors'],
    }
    const paperData = {
        name: 'paper',
        label: 'Feuille',
        icon: 'âœ‹',
        strongerThan: ['rock'],
    }
    const scissorsData = {
        name: 'scissors',
        label: 'Ciseaux',
        icon: 'âœŒ',
        strongerThan: ['paper'],
    }

    const tree = renderer
        .create(
            <div>
                <Item item={rockData} id={1} />
                <Item item={paperData} id={2} />
                <Item item={scissorsData} id={3} />
            </div>
        )
        .toJSON()
    expect(tree).toMatchSnapshot()
})
