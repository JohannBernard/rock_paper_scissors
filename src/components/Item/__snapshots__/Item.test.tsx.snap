// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches item snapshot 1`] = `
<div>
  <div
    className="my_coach_item"
    data-testid="testItem-1"
  >
    <p
      className="my_coach_item_icon"
      data-testid="testItemIcon-1"
    >
      ðŸ‘Š
    </p>
    <p
      className="my_coach_item_label"
      data-testid="testItemLabel-1"
    >
      Pierre
    </p>
  </div>
  <div
    className="my_coach_item"
    data-testid="testItem-2"
  >
    <p
      className="my_coach_item_icon"
      data-testid="testItemIcon-2"
    >
      âœ‹
    </p>
    <p
      className="my_coach_item_label"
      data-testid="testItemLabel-2"
    >
      Feuille
    </p>
  </div>
  <div
    className="my_coach_item"
    data-testid="testItem-3"
  >
    <p
      className="my_coach_item_icon"
      data-testid="testItemIcon-3"
    >
      âœŒ
    </p>
    <p
      className="my_coach_item_label"
      data-testid="testItemLabel-3"
    >
      Ciseaux
    </p>
  </div>
</div>
`;
